src/Datadog.Trace.ClrProfiler.Managed/Integrations/Caching/Interception.cs

```diff
@@ -23,20 +23,20 @@ internal static Type[] TypeArray(params object[] objectsToCheck)
             return types;
         }
 
-        internal static string MethodKey(Type[] genericTypes, Type[] parameterTypes)
+        internal static string MethodKey(Type returnType, Type[] genericTypes, Type[] parameterTypes)
         {
-            var key = "m";
+            var key = $"m_r{returnType.AssemblyQualifiedName}";
 
             for (int i = 0; i < (genericTypes?.Length ?? 0); i++)
             {
                 Debug.Assert(genericTypes != null, nameof(genericTypes) + " != null");
-                key = string.Concat(key, $"_g{genericTypes[i].FullName}");
+                key = string.Concat(key, $"_g{genericTypes[i].AssemblyQualifiedName}");
```

Comment:
Looking at the docs here, I think FullName was what we actually wanted, which includes the <AssemblyName>.<NamespaceName>.<TypeName>: https://docs.microsoft.com/en-us/dotnet/api/system.type.assemblyqualifiedname?view=netframework-4.8

I think we should revert this. Until I checked this, I thought FullName would exclude the AssemblyName but that appears to be incorrect.